modules/apps/okular.nix

```
@@ -43,6 +43,53 @@ in with lib.types;
         default = null;
         type = nullOr bool;
       };
+
+      viewMode = lib.mkOption {
+        description = "The view mode for the pages.";
+        default = "Single";
+        type =
+          nullOr (enum [ "Single" "Facing" "FacingFirstCentered" "Summary" ]);
+      };
+
+      zoomMode = lib.mkOption {
+        description = ''
+          Specifies the default zoom mode for file which were never opened before. For those files which were opened before the previous zoom mode is applied.
+          0: "100%"
+          1: "Fit Width"
+          2: "Fit Page"
+          3: "Auto Fit"
+          '';
+        default = 1;
+        type = nullOr (enum [ 0 1 2 3 ]);
+      };
```

Default should be set to `null`
Use getIndexFromEnum function for zoomMode:
```suggestion
      zoomMode =
        let enumVals = [ "100%" "fitWidth" "fitPage" "autoFit" ];
        in lib.mkOption {
          description = ''
            Specifies the default zoom mode for file which were never opened before. For those files which were opened before the previous zoom mode is applied.
            '';
          default = null;
          type = nullOr (enum enumVals);
          apply = getIndexFromEnum enumVals;
        };
```

Add this function in the let statement in the start of the file.
```nix
getIndexFromEnum = enum: value:
  if value == null
  then null
  else
    lib.lists.findFirstIndex
      (x: x == value)
      (throw "getIndexFromEnum (okular): Value ${value} isn't present in the enum. This is a bug")
      enum;
```

---

Thank you! I want to do something like this but didn't know how to go about it. I was hoping that there is some kind of `enum` that allows setting both a value and display string.

---

You could use the `oneOf` function for that. For example:

```nix
type = nullOr (oneOf (enum enumVals) (enum [0 1 2 3]));
```

But you would need to modify the apply function to be like this:

```nix
apply = zoomMode: if builtins.isString zoomMode then getIndexFromEnum enumVals zoomMode else zoomMode
```